zone/raids.cpp

```diff
@@ -1543,21 +1545,22 @@ void Raid::SendRaidGroupRemove(const char *who, uint32 gid)
 
 void Raid::SendRaidMOTD(Client *c)
 {
-	if (!c || motd.empty()) {
+	if (!c || motd.empty() || c->IsBot()) {
 		return;
 	}
 
-	if (entity_list.GetBotByBotName(c->GetName())) {
-		return;
-	}
+	auto outapp = new EQApplicationPacket(OP_RaidUpdate, sizeof(RaidMOTD_Struct));
+	auto data = (RaidMOTD_Struct*)outapp->pBuffer;
 
-	size_t size = motd.size() + 1;
-	auto outapp = new EQApplicationPacket(OP_RaidUpdate, sizeof(RaidMOTD_Struct) + size);
-	auto rmotd = (RaidMOTD_Struct *)outapp->pBuffer;
-	rmotd->general.action = raidSetMotd;
-	strn0cpy(rmotd->general.player_name, c->GetName(), 64);
-	strn0cpy(rmotd->motd, motd.c_str(), size);
-	c->FastQueuePacket(&outapp);
+	data->general.action = raidSetMotd;
+	data->general.parameter = 0;
+	data->general.unknown1 = 0;
+	strncpy(data->general.leader_name, c->GetName(), sizeof(c->GetName()));
+	strncpy(data->general.player_name, GetLeaderName().c_str(), 64);
+	strncpy(data->motd, motd.c_str(), sizeof(data->motd));
```

Comment:
All the regular strncpy calls should be changed to `strn0cpy` (ones in translators too that were renamed from strncpy_s)

Reply:
Good morning.  Updated raids.cpp (including another spot), translators, and the defines for OUT_str.